FA/fa/Finite_Automata.py

```diff
@@ -0,0 +1,96 @@
+import sys
+from termcolor import colored, cprint
+
+class Finite_Automata:
+    def __init__(self, file_path):
+        self.file = file_path
+        self.fa = []
+        self.start_state = None
+        self.final_states = []
+        self.alphabet = None
+        self.states = None
+        self.transitions = {}
+        self.current_state = 'A'
+        self.read_file()
+
+    def read_file(self):
+        file = open(self.file, 'r')
+        fa = []
+        for line in file:
+            line.replace("\n","")
+            fa.append(line)
```

Comment:
I don't quite understand the need for self.fa in this context, seems a bit redundant to me ....

Reply:
But it may be that I'm just bad a reading other people's code

Reply:
You are right, I didn't use `self.fa` and forgot to erase it.
I use fa to have my file content in a list to resemble more the structure of the file as in `fa[1]` will be the second row from the file. I find it easier for later, e.g. i know that the alphabet is found on the second line so I will have it in `fa[1]`